pixel-saver@deadalnix.me/buttons.js

```diff
@@ -231,7 +262,20 @@ function init(extensionMeta) {
 
 let wmCallbackIDs = [];
 let overviewCallbackIDs = [];
-function enable() {
+let options;
+
+function loadOptions() {
+	options = new Settings.Options();
+	options.BUTTON_POSITION.changed(Lang.bind(this, function() {
+		if (options.BUTTON_POSITION.get() == Position.HIDDEN) {
+			hideButtons();
+		} else {
+			showButtons();
+		}
+	}));
+}
```

Comment:
As per your request, the wiring now happens in the buttons module. 

If I understood you correctly, the options should also be loaded here. This implies that we have to perform cleanup as well. Generally not a problem, but as more modules use the options, they all have to perform cleanup if they listen to settings changes.

I would therefore suggest that at least the option loading should move to the Extension module as then cleanup has to performed only once (and you don't have to remember the constraint to cleanup options if you listen for changes).